includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,554 @@
+			);
+
+			// Localize the script
+			$translation_array = array(
+				'ajax_url'  => admin_url( 'admin-ajax.php' ),
+				'text_wait' => __( 'Please wait...', 'swedbank-pay-woocommerce-payments' ),
+			);
+			wp_localize_script( 'swedbank-pay-admin-js', 'SwedbankPay_Admin', $translation_array );
+
+			// Enqueued script with localized data
+			wp_enqueue_script( 'swedbank-pay-admin-js' );
+		}
+	}
+
+	/**
+	 * Action for Capture
+	 */
+	public function ajax_swedbank_pay_capture() {
+		if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'swedbank_pay' ) ) {
+			exit( 'No naughty business' );
+		}
+
+		$order_id = (int) $_REQUEST['order_id'];
+		$order    = wc_get_order( $order_id );
+
+		// Get Payment Gateway
+		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+		if ( isset( $gateways[ $order->get_payment_method() ] ) ) {
+			$gateway = $gateways[ $order->get_payment_method() ];
+
+			try {
+				$gateway->capture_payment( $order_id );
+				wp_send_json_success( __( 'Capture success.', 'swedbank-pay-woocommerce-payments' ) );
+			} catch ( Exception $e ) {
+				$message = $e->getMessage();
+				wp_send_json_error( $message );
+			}
+		}
+	}
+
+	/**
+	 * Action for Cancel
+	 */
+	public function ajax_swedbank_pay_cancel() {
+		if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'swedbank_pay' ) ) {
+			exit( 'No naughty business' );
+		}
+
+		$order_id = (int) $_REQUEST['order_id'];
+		$order    = wc_get_order( $order_id );
+
+		// Get Payment Gateway
+		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+		if ( isset( $gateways[ $order->get_payment_method() ] ) ) {
+			$gateway = $gateways[ $order->get_payment_method() ];
+
+			try {
+				$gateway->cancel_payment( $order_id );
+				wp_send_json_success( __( 'Cancel success.', 'swedbank-pay-woocommerce-payments' ) );
+			} catch ( Exception $e ) {
+				$message = $e->getMessage();
+				wp_send_json_error( $message );
+			}
+		}
+	}
+
+	/**
+	 * Generate UUID
+	 *
+	 * @param $node
+	 *
+	 * @return string
+	 */
+	public function generate_uuid( $node ) {
+		return \Ramsey\Uuid\Uuid::uuid5( \Ramsey\Uuid\Uuid::NAMESPACE_OID, $node )->toString();
+	}
+
+	/**
+	 * Dispatch Background Process
+	 */
+	public function maybe_process_queue() {
+		self::$background_process->dispatch();
+	}
+
+	/**
+	 * Provide Admin Menu items
+	 */
+	public function admin_menu() {
+		// Add Upgrade Page
+		global $_registered_pages;
+
+		$hookname = get_plugin_page_hookname( 'wc-payex-psp-upgrade', '' );
+		if ( ! empty( $hookname ) ) {
+			add_action( $hookname, __CLASS__ . '::upgrade_page' );
+		}
+
+		$_registered_pages[ $hookname ] = true;
+	}
+
+	/**
+	 * Upgrade Page
+	 */
+	public static function upgrade_page() {
+		if ( ! current_user_can( 'update_plugins' ) ) {
+			return;
+		}
+
+		// Run Database Update
+		include_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-update.php' );
+		\WC_Swedbank_Pay_Update::update();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\WC_Swedbank_Pay_Update' in method 'upgrade_page'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)